lms/djangoapps/course_api/blocks/transformers/tests/test_milestones.py

```diff
@@ -171,3 +190,55 @@ def test_exam_created(self, exam_ref, attempt_status, expected_blocks):
             set(block_structure.get_block_keys()),
             set(self.get_block_key_set(self.blocks, *expected_blocks)),
         )
+
+    def test_special_exam_gated(self):
+        expected_blocks = ('course', 'A', 'B', 'C', 'TimedExam', 'D', 'E', 'NotASpecialExam', 'H')
+        self.setup_gated_section(self.blocks['PracticeExam'], self.blocks['TimedExam'])
+        block_structure = get_course_blocks(
+            self.user,
+            self.course.location,
+            self.transformers,
+        )
+        self.assertEqual(
+            set(block_structure.get_block_keys()),
+            set(self.get_block_key_set(self.blocks, *expected_blocks)),
+        )
+
+    def test_not_special_exam_gated(self):
+        expected_blocks = ('course', 'A', 'B', 'C', 'TimedExam', 'D', 'E', 'PracticeExam', 'F', 'G')
+        self.setup_gated_section(self.blocks['NotASpecialExam'], self.blocks['TimedExam'])
```

Comment:
Update this test to make sure the gated content becomes available after the `TimedExam` is "completed".